clouddriver-ecs/src/integration/java/com/netflix/spinnaker/clouddriver/ecs/test/CreateServerGroupWithArtifactsSpec.java

```diff
@@ -319,4 +320,62 @@ public void createServerGroup_ArtifactsFARGATETgMappingsTest()
     assertEquals(80, serviceLB.getContainerPort().intValue());
     assertEquals("integArtifactsFargateTgMappings-cluster", seenCreateServRequest.getCluster());
   }
+
+  @DisplayName(
+      ".\n===\n"
+          + "Given description w/ task def artifacts, EC2 launch type, and new target group fields "
+          + "without container definition, gives an exception(Provided task definition does not contain any container definitions). "
+          + "\n===")
+  @Test
+  public void createServerGroup_errorIfNoContainersTest() throws IOException, InterruptedException {
+
+    // given
+    String url = getTestUrl(CREATE_SG_TEST_PATH);
+    String requestBody =
+        generateStringFromTestFile("/createServerGroup-artifact-EC2-targetGroupMappings.json");
+
+    setEcsAccountCreds();
+
+    ByteArrayInputStream byteArrayInputStreamOfArtifactsForEC2Type =
+        new ByteArrayInputStream(
+            generateStringFromTestArtifactFile(
+                    "createServerGroup-artifact-EC2-targetGroup-WithNoContainers-artifactFile.json")
+                .getBytes());
+
+    when(mockArtifactDownloader.download(any(Artifact.class)))
+        .thenReturn(byteArrayInputStreamOfArtifactsForEC2Type);
+
+    String taskId =
+        given()
+            .contentType(ContentType.JSON)
+            .body(requestBody)
+            .when()
+            .post(url)
+            .then()
+            .statusCode(200)
+            .contentType(ContentType.JSON)
+            .body("id", notNullValue())
+            .body("resourceUri", containsString("/task/"))
+            .extract()
+            .path("id");
+
+    retryUntilTrue(
+        () -> {
+          HashMap<String, Object> status =
+              get(getTestUrl("/task/" + taskId))
+                  .then()
+                  .contentType(ContentType.JSON)
+                  .extract()
+                  .path("status");
+          assertEquals(
+              true,
+              status
+                  .get("status")
+                  .toString()
+                  .contains("Provided task definition does not contain any container definitions"));
+          return status.get("failed").equals(true);
+        },
+        String.format("Failed to detect service creation in %s seconds", TASK_RETRY_SECONDS),
```

Comment:
This msg should indicate we failed to detect _task failure_, since this is checking for an error case

Reply:
Resolved.